test_conformance/images/kernel_read_write/test_cl_ext_image_from_buffer.cpp

```diff
@@ -0,0 +1,923 @@
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    if (!is_extension_available(device, "cl_ext_image_from_buffer"))
+    {
+        printf("Extension cl_ext_image_from_buffer not available");
+        return TEST_SKIPPED_ITSELF;
+    }
+
+    std::vector<cl_mem_object_type> imageTypes{
+        CL_MEM_OBJECT_IMAGE1D,       CL_MEM_OBJECT_IMAGE2D,
+        CL_MEM_OBJECT_IMAGE3D,       CL_MEM_OBJECT_IMAGE1D_BUFFER,
+        CL_MEM_OBJECT_IMAGE1D_ARRAY, CL_MEM_OBJECT_IMAGE2D_ARRAY
+    };
+
+    for (auto imageType : imageTypes)
+    {
+        /* Get the list of supported image formats */
+        std::vector<cl_image_format> formatList;
+        if (TEST_PASS
+                != get_format_list(context, imageType, formatList,
+                                   CL_MEM_READ_WRITE)
+            || formatList.size() == 0)
+        {
+            test_fail("Failure to get supported formats list");
+        }
+
+        for (auto format : formatList)
+        {
+            cl_image_desc image_desc = { 0 };
+            image_desc_init(&image_desc, imageType);
+
+            size_t row_pitch_alignment = 0;
+            size_t slice_pitch_alignment = 0;
+            size_t base_address_alignment = 0;
+
+            int get_error = get_image_requirement_alignment(
+                device, context, CL_MEM_READ_WRITE, &format, &image_desc,
+                &row_pitch_alignment, &slice_pitch_alignment,
+                &base_address_alignment);
+            if (TEST_PASS != get_error)
+            {
+                return get_error;
+            }
+
+            const size_t element_size =
+                get_format_size(context, &format, imageType);
+
+            const size_t row_pitch = aligned_size(
+                TEST_IMAGE_SIZE * element_size, row_pitch_alignment);
+            const size_t slice_pitch = aligned_size(row_pitch * TEST_IMAGE_SIZE,
+                                                    slice_pitch_alignment);
+
+            const size_t buffer_size = (slice_pitch + 1)
+                * TEST_IMAGE_SIZE; /* For bigger row/slice pitch */
+
+            cl_int err = CL_SUCCESS;
+            cl_mem buffer = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                           buffer_size, nullptr, &err);
+            test_error(err, "Unable to create buffer");
+
+            /* Test Row pitch images */
+            if (imageType == CL_MEM_OBJECT_IMAGE2D
+                || imageType == CL_MEM_OBJECT_IMAGE3D
+                || imageType == CL_MEM_OBJECT_IMAGE1D_ARRAY
+                || imageType == CL_MEM_OBJECT_IMAGE2D_ARRAY)
+            {
+                image_desc.buffer = buffer;
+                image_desc.image_row_pitch =
+                    row_pitch + 1; /* wrong row pitch */
+
+                clCreateImage(context, CL_MEM_READ_WRITE, &format, &image_desc,
+                              nullptr, &err);
+                test_failure_error(err, CL_INVALID_IMAGE_FORMAT_DESCRIPTOR,
```

Comment:
```suggestion
                test_failure_error(err, CL_INVALID_IMAGE_DESCRIPTOR,
```

Reply:
Fixing that.

Reply:
According to the spec, this should indeed be CL_INVALID_IMAGE_FORMAT_DESCRIPTOR. Can we clarify, and if correct, fix the comment at the top of this function?

https://github.com/KhronosGroup/OpenCL-Docs/pull/752/files#diff-8eb35a0189bdc7abf4bbab8a4481cf791a5331dddbe2ec9dc40ea7a0cc1bfc19R97

Reply:
I was thinking about changing the specification of the new extension as this is related to  cl_image_desc values.
But in the existing specification we already use CL_INVALID_IMAGE_FORMAT_DESCRIPTOR for similar cases for 2D images : https://www.khronos.org/registry/OpenCL/sdk/2.0/docs/man/xhtml/clCreateImage.html

We then need to keep that as it is.  @Jeremy-Kemp Do you agree on that ?

Reply:
Agreed. Keep CL_INVALID_IMAGE_FORMAT_DESCRIPTOR to match the base OpenCL spec.